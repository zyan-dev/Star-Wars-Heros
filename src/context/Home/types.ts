import { User } from '../../types/Home';

export const SET_RESULT = 'Set fetched user';
export const SEARCHING_RESULT = 'Searching people';
export const SET_PAGE = 'Set page number';
export const SET_MAX_PAGE = 'Set max page number';

export type SetResultAction = {
  type: typeof SET_RESULT;
  payload: User[];
};

export type SearchResultAction = {
  type: typeof SEARCHING_RESULT;
  payload: boolean;
};

export type SetPageAction = {
  type: typeof SET_PAGE;
  payload: number;
}
  ;
export type SetMaxPageAction = {
  type: typeof SET_MAX_PAGE;
  payload: number;
};

export type HomeAction = (
  SetResultAction |
  SearchResultAction |
  SetPageAction |
  SetMaxPageAction
);
