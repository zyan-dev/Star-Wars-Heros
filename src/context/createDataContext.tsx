import React, { useReducer } from 'react';

export type ContextParam = {
    children: Array<React.ReactNode>
}

// context function for whole app state management
export default (reducer: any, actions: any, defaultValue: any) => {
  const Context = React.createContext(defaultValue);

  // provider component wrapping all components
  const Provider = (param: any) => {
    const { children, value } = param;
    const [state, dispatch] = useReducer(reducer, { ...defaultValue, ...value });

    const boundActions: any = {};

    // action objects looped over and bound to dispatch
    for (const key in actions) {
      boundActions[key] = actions[key](dispatch);
    }

    return (
    // bound actions change state
      <Context.Provider value={{ state, ...boundActions }}>
        {children}
      </Context.Provider>
    );
  };

  // return context and provider
  return { Context, Provider };
};
