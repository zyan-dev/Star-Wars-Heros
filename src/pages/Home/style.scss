.home-page {
  width: 100%;
  .search-popup {
    max-width: 600px;
    width: 90%;
    height: 600px;
    margin: 100px auto;
    box-shadow: 10px 10px 20px grey;
    .header {
      background-color: royalblue;
      padding: 20px;
      p {
        text-align: center;
        margin: 0;
      }
    }
    .search-input-wrapper {
      margin: 15px;
      border: 1px solid black;
      padding: 15px;
      background-color: darkgrey;
      p {
        margin: 0;
        margin-bottom: 5px;
      }
      .search-input {
        display: flex;
        align-items: center;
        input {
          flex: 1;
          padding: 4px;
        }
        button {
          margin-left: 15px;
          background-color: royalblue;
          height: 27px;
          padding: 0 20px;
          &:hover {
            outline: none;
          }
        }
      }
    }
    .result-view-wrapper {
      border: 1px solid black;
      margin: 0 15px;
      height: 320px;
      overflow-y: auto;
      .status {
        margin-top: 100px;
        text-align: center;
      }
      .info-line {
        display: flex;
        padding: 8px 25px;
        justify-content: space-between;
        align-items: center;
        border-top: 2px solid black;
      }
      .name {
        background-color: darkgrey;
        margin: 0;
        padding: 15px;
        font-weight: 600;
      }
    }
    .page-controller {
      display: flex;
      justify-content: center;
      margin-top: 30px;
      button {
        margin: 0 20px;
      }
    }
  }
}
