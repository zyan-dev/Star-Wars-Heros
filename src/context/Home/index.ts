import { User } from '../../types/Home';
import createDataContext from '../createDataContext';
import {
  HomeAction,
  SET_RESULT,
  SEARCHING_RESULT,
  SET_PAGE,
  SET_MAX_PAGE,
} from './types';

export interface HomeState {
  page: number;
  peoples: User[] | string;
  searching: boolean;
  maxPage: number;
}

const initialState: HomeState = {
  page: 1,
  peoples: '',
  searching: false,
  maxPage: 1,
};

const homeReducer = (state: HomeState, action: HomeAction) => {
  switch (action.type) {
    case SET_RESULT:
      return {
        ...state,
        peoples: action.payload,
      };
    case SEARCHING_RESULT:
      return {
        ...state,
        searching: action.payload,
      };
    case SET_PAGE:
      return {
        ...state,
        page: action.payload,
      };
    case SET_MAX_PAGE:
      return {
        ...state,
        maxPage: action.payload,
      };
    default:
      return state;
  }
};

export const dispatchAction = (dispatch: any) => async (action: HomeAction) => {
  dispatch(action);
};

export const searchPeople = (dispatch: any) => (name: string, page: number) => {
  dispatch({ type: SEARCHING_RESULT, payload: true });
  fetch(`https://swapi.dev/api/people?search=${name}&page=${page}`, {
    method: 'GET',
    headers: {
      // 'Authentication': 'Bearer ac6ce528245439fda03b3539c60427d2',
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
  })
    .then((res) => res.json())
    .then((res) => {
      if (res.results && res.results.length > 0) {
        dispatch({ type: SET_RESULT, payload: res.results });
        const mp = Math.floor(res.count / 10) + 1;
        dispatch({ type: SET_MAX_PAGE, payload: mp });
        if (mp < page) dispatch({ type: SET_PAGE, payload: 1 });
      } else {
        dispatch({ type: SET_RESULT, payload: 'Not found' });
      }
    })
    .catch((e) => {
      console.error(e.toString());
      dispatch({ type: SET_RESULT, payload: 'Error occured' });
    })
    .finally(() => {
      dispatch({ type: SEARCHING_RESULT, payload: false });
    });
};

export const { Provider, Context } = createDataContext(
  homeReducer,
  {
    dispatchAction,
    searchPeople,
  },
  initialState,
);
