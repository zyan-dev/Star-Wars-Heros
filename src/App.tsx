import React from 'react';
import { Provider as HomeProvider } from './context/Home';
import './App.scss';
import AppRouter from './navigations/Routers';

function App() {
  return (
    <HomeProvider>
      <AppRouter />
    </HomeProvider>
  );
}

export default App;
