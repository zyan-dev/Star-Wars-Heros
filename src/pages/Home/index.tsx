import React, { useContext, useEffect, useState } from 'react';
import { Context as HomeContext } from '../../context/Home';
import { SET_PAGE } from '../../context/Home/types';
import { User } from '../../types/Home';
import { captializeFirstLetter } from '../../utils/helpers';
import './style.scss';

const HomePage: React.FC = () => {
  const [name, setName] = useState('');
  const {
    dispatchAction,
    searchPeople,
    state: {
      page,
      peoples,
      searching,
      maxPage,
    },
  } = useContext(HomeContext);

  useEffect(() => {
    if (name.length > 0) searchPeople(name, page);
  }, [page]);

  const handleNameChange = (event: React.FormEvent<HTMLInputElement>) => {
    setName(event.currentTarget.value);
  };

  const onPressSearch = () => {
    searchPeople(name, page);
  };

  const onPressPrevPage = () => {
    dispatchAction({ type: SET_PAGE, payload: page - 1 });
  };

  const onPressNextPage = () => {
    dispatchAction({ type: SET_PAGE, payload: page + 1 });
  };

  return (
    <div className="home-page">
      <div className="search-popup">
        <div className="header">
          <p>STAR WARS HEROES</p>
        </div>
        <div className="search-input-wrapper">
          <p>Character Name</p>
          <div className="search-input">
            <input data-testid="search-input" aria-label="search-input" type="input" value={name} onChange={handleNameChange} />
            <button data-testid="search-submit" type="button" onClick={onPressSearch}>SEARCH</button>
          </div>
        </div>
        <div className="result-view-wrapper">
          {searching && (
            <p className="status">Searching...</p>
          )}
          {!searching && (typeof peoples === 'string') && (
            <p className="status">{peoples}</p>
          )}
          {!searching && (typeof peoples !== 'string') && peoples.length > 0 && peoples.map((user: User, index: number) => (
            <div key={user.url}>
              <p className="name">{index + 1}. {user.name}</p>
              <div className="info-line">
                <span>Gender</span>
                <span data-testid="gender">{captializeFirstLetter(user.gender)}</span>
              </div>
              <div className="info-line">
                <span>Birth Year</span>
                <span data-testid="birth_year">{user.birth_year}</span>
              </div>
              <div className="info-line">
                <span>Height</span>
                <span data-testid="height">{user.height}</span>
              </div>
              <div className="info-line">
                <span>Mass</span>
                <span data-testid="mass">{user.mass}</span>
              </div>
              <div className="info-line">
                <span>Hair Color</span>
                <span data-testid="hair_color">{user.hair_color}</span>
              </div>
            </div>
          ))}
        </div>
        <div className="page-controller">
          <button data-testid="prev-page" type="button" onClick={onPressPrevPage} disabled={page === 1}>{'<'}</button>
          <span data-testid="page">Page {page}</span>
          <button data-testid="next-page" type="button" onClick={onPressNextPage} disabled={page === maxPage}>{'>'}</button>
        </div>
      </div>
    </div>
  );
};

export default HomePage;
