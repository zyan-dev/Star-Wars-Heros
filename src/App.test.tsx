import React from 'react';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import {HomeState, Provider as HomeProvider} from './context/Home';
import HomePage from './pages/Home';
import { User } from './types/Home';
import { HomeAction, SEARCHING_RESULT, SET_MAX_PAGE, SET_PAGE, SET_RESULT } from './context/Home/types';
import createDataContext from './context/createDataContext';

export const delayFunc = (callback: any) => {
  setTimeout(() => {
    callback && callback();
  }, 1000);
}

describe('Testing home page', () => {
  let element;
  let wrapper;

  test('renders search input and buttons correctly', () => {
    wrapper = render(<App />);
    element = screen.getByText(/STAR WARS HEROES/i);
    expect(element).toBeInTheDocument();
    element = screen.getByText(/Character Name/i);
    expect(element).toBeInTheDocument();
    element = screen.getByText(/SEARCH/i);
    expect(element).toBeInTheDocument();
  });

  cleanup();

  test('render filtered people correctly', () => {
    const customRender = (ui: React.ReactNode, providerProps: any) => {
      return render(
        <HomeProvider value={providerProps}>{ui}</HomeProvider>
      )
    }
    const peoples: User[] = [
      {
        "birth_year": "19 BBY",
        "eye_color": "Blue",
        "films": [
            "https://swapi.dev/api/films/1/",
        ],
        "gender": "Male",
        "hair_color": "Blond",
        "height": "172",
        "homeworld": "https://swapi.dev/api/planets/1/",
        "mass": "77",
        "name": "Luke Skywalker",
        "skin_color": "Fair",
        "created": "2014-12-09T13:50:51.644000Z",
        "edited": "2014-12-10T13:52:43.172000Z",
        "species": [
            "https://swapi.dev/api/species/1/"
        ],
        "starships": [
            "https://swapi.dev/api/starships/12/",
        ],
        "url": "https://swapi.dev/api/people/1/",
        "vehicles": [
            "https://swapi.dev/api/vehicles/14/"
        ]
      },
      {
        "birth_year": "600BBY",
        "created": "2014-12-10T17:11:31.638000Z",
        "edited": "2014-12-20T21:17:50.338000Z",
        "eye_color": "orange",
        "films": ["https://swapi.dev/api/films/1/"],
        "gender": "hermaphrodite",
        "hair_color": "n/a",
        "height": "175",
        "homeworld": "https://swapi.dev/api/planets/24/",
        "mass": "1,358",
        "name": "Jabba Desilijic Tiure",
        "skin_color": "green-tan, brown",
        "species": ["https://swapi.dev/api/species/5/"],
        "starships": [],
        "url": "https://swapi.dev/api/people/16/",
        "vehicles": []
      }
    ];
    const providerProps: HomeState = {
      page: 1,
      searching: false,
      peoples,
      maxPage: 1
    }
    wrapper = customRender(
      <HomePage />,
      providerProps
    );
    element = screen.getByText('2. Jabba Desilijic Tiure');
    expect(element).toBeInTheDocument();
    element = screen.getByText('1. Luke Skywalker');
    expect(element).toBeInTheDocument();
    expect(screen.getAllByTestId('gender').length).toBe(2);
    expect(screen.getAllByTestId('birth_year').length).toBe(2);
    expect(screen.getAllByTestId('height').length).toBe(2);
    expect(screen.getAllByTestId('mass').length).toBe(2);
    expect(screen.getAllByTestId('hair_color').length).toBe(2);
  });

  test('HomePage events works correctly', async () => {
    const customRender = (ui: React.ReactNode, providerProps: any) => {
      return render(
        <HomeProvider value={providerProps}>{ui}</HomeProvider>
      )
    }
    const providerProps: HomeState = {
      page: 2,
      searching: false,
      peoples: [],
      maxPage: 3
    }
    wrapper = customRender(
      <HomePage />,
      providerProps
    );
    element = screen.getByTestId('search-input');
    fireEvent.change(element, { target: { value: 'ab' } });
    element = screen.getByLabelText('search-input') as HTMLInputElement;
    expect(element.value).toBe('ab');
    element = screen.getByTestId('search-submit');
    fireEvent.click(element);
    element = screen.getByTestId('prev-page');
    fireEvent.click(element);
    element = screen.getByTestId('next-page');
    fireEvent.click(element);
    expect(screen.getByTestId('page').textContent).toBe('Page 2');
  })
})



