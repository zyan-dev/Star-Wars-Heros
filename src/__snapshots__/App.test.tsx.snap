// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing home page HomeProvider works correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-page"
      >
        <div
          class="search-popup"
        >
          <div
            class="header"
          >
            <p>
              STAR WARS HEROES
            </p>
          </div>
          <div
            class="search-input-wrapper"
          >
            <p>
              Character Name
            </p>
            <div
              class="search-input"
            >
              <input
                data-testid="search-input"
                type="input"
                value=""
              />
              <button
                data-testid="search-submit"
                type="button"
              >
                SEARCH
              </button>
            </div>
          </div>
          <div
            class="result-view-wrapper"
          >
            <p
              class="status"
            />
          </div>
          <div
            class="page-controller"
          >
            <button
              data-testid="prev-page"
              disabled=""
              type="button"
            >
              &lt;
            </button>
            <span
              data-testid="page-1"
            >
              Page 
              1
            </span>
            <button
              data-testid="next-page"
              disabled=""
              type="button"
            >
              &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-page"
    >
      <div
        class="search-popup"
      >
        <div
          class="header"
        >
          <p>
            STAR WARS HEROES
          </p>
        </div>
        <div
          class="search-input-wrapper"
        >
          <p>
            Character Name
          </p>
          <div
            class="search-input"
          >
            <input
              data-testid="search-input"
              type="input"
              value=""
            />
            <button
              data-testid="search-submit"
              type="button"
            >
              SEARCH
            </button>
          </div>
        </div>
        <div
          class="result-view-wrapper"
        >
          <p
            class="status"
          />
        </div>
        <div
          class="page-controller"
        >
          <button
            data-testid="prev-page"
            disabled=""
            type="button"
          >
            &lt;
          </button>
          <span
            data-testid="page-1"
          >
            Page 
            1
          </span>
          <button
            data-testid="next-page"
            disabled=""
            type="button"
          >
            &gt;
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing home page render filtered people correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-page"
      >
        <div
          class="search-popup"
        >
          <div
            class="header"
          >
            <p>
              STAR WARS HEROES
            </p>
          </div>
          <div
            class="search-input-wrapper"
          >
            <p>
              Character Name
            </p>
            <div
              class="search-input"
            >
              <input
                data-testid="search-input"
                type="input"
                value=""
              />
              <button
                data-testid="search-submit"
                type="button"
              >
                SEARCH
              </button>
            </div>
          </div>
          <div
            class="result-view-wrapper"
          >
            <div>
              <p
                class="name"
              >
                1
                . 
                Luke Skywalker
              </p>
              <div
                class="info-line"
              >
                <span>
                  Gender
                </span>
                <span
                  data-testid="gender"
                >
                  Male
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Birth Year
                </span>
                <span
                  data-testid="birth_year"
                >
                  19 BBY
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Height
                </span>
                <span
                  data-testid="height"
                >
                  172
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Mass
                </span>
                <span
                  data-testid="mass"
                >
                  77
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Hair Color
                </span>
                <span
                  data-testid="hair_color"
                >
                  Blond
                </span>
              </div>
            </div>
            <div>
              <p
                class="name"
              >
                2
                . 
                Jabba Desilijic Tiure
              </p>
              <div
                class="info-line"
              >
                <span>
                  Gender
                </span>
                <span
                  data-testid="gender"
                >
                  Hermaphrodite
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Birth Year
                </span>
                <span
                  data-testid="birth_year"
                >
                  600BBY
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Height
                </span>
                <span
                  data-testid="height"
                >
                  175
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Mass
                </span>
                <span
                  data-testid="mass"
                >
                  1,358
                </span>
              </div>
              <div
                class="info-line"
              >
                <span>
                  Hair Color
                </span>
                <span
                  data-testid="hair_color"
                >
                  n/a
                </span>
              </div>
            </div>
          </div>
          <div
            class="page-controller"
          >
            <button
              data-testid="prev-page"
              disabled=""
              type="button"
            >
              &lt;
            </button>
            <span
              data-testid="page-1"
            >
              Page 
              1
            </span>
            <button
              data-testid="next-page"
              disabled=""
              type="button"
            >
              &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-page"
    >
      <div
        class="search-popup"
      >
        <div
          class="header"
        >
          <p>
            STAR WARS HEROES
          </p>
        </div>
        <div
          class="search-input-wrapper"
        >
          <p>
            Character Name
          </p>
          <div
            class="search-input"
          >
            <input
              data-testid="search-input"
              type="input"
              value=""
            />
            <button
              data-testid="search-submit"
              type="button"
            >
              SEARCH
            </button>
          </div>
        </div>
        <div
          class="result-view-wrapper"
        >
          <div>
            <p
              class="name"
            >
              1
              . 
              Luke Skywalker
            </p>
            <div
              class="info-line"
            >
              <span>
                Gender
              </span>
              <span
                data-testid="gender"
              >
                Male
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Birth Year
              </span>
              <span
                data-testid="birth_year"
              >
                19 BBY
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Height
              </span>
              <span
                data-testid="height"
              >
                172
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Mass
              </span>
              <span
                data-testid="mass"
              >
                77
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Hair Color
              </span>
              <span
                data-testid="hair_color"
              >
                Blond
              </span>
            </div>
          </div>
          <div>
            <p
              class="name"
            >
              2
              . 
              Jabba Desilijic Tiure
            </p>
            <div
              class="info-line"
            >
              <span>
                Gender
              </span>
              <span
                data-testid="gender"
              >
                Hermaphrodite
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Birth Year
              </span>
              <span
                data-testid="birth_year"
              >
                600BBY
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Height
              </span>
              <span
                data-testid="height"
              >
                175
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Mass
              </span>
              <span
                data-testid="mass"
              >
                1,358
              </span>
            </div>
            <div
              class="info-line"
            >
              <span>
                Hair Color
              </span>
              <span
                data-testid="hair_color"
              >
                n/a
              </span>
            </div>
          </div>
        </div>
        <div
          class="page-controller"
        >
          <button
            data-testid="prev-page"
            disabled=""
            type="button"
          >
            &lt;
          </button>
          <span
            data-testid="page-1"
          >
            Page 
            1
          </span>
          <button
            data-testid="next-page"
            disabled=""
            type="button"
          >
            &gt;
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing home page renders search input and buttons correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-page"
      >
        <div
          class="search-popup"
        >
          <div
            class="header"
          >
            <p>
              STAR WARS HEROES
            </p>
          </div>
          <div
            class="search-input-wrapper"
          >
            <p>
              Character Name
            </p>
            <div
              class="search-input"
            >
              <input
                data-testid="search-input"
                type="input"
                value=""
              />
              <button
                data-testid="search-submit"
                type="button"
              >
                SEARCH
              </button>
            </div>
          </div>
          <div
            class="result-view-wrapper"
          >
            <p
              class="status"
            />
          </div>
          <div
            class="page-controller"
          >
            <button
              data-testid="prev-page"
              disabled=""
              type="button"
            >
              &lt;
            </button>
            <span
              data-testid="page-1"
            >
              Page 
              1
            </span>
            <button
              data-testid="next-page"
              disabled=""
              type="button"
            >
              &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-page"
    >
      <div
        class="search-popup"
      >
        <div
          class="header"
        >
          <p>
            STAR WARS HEROES
          </p>
        </div>
        <div
          class="search-input-wrapper"
        >
          <p>
            Character Name
          </p>
          <div
            class="search-input"
          >
            <input
              data-testid="search-input"
              type="input"
              value=""
            />
            <button
              data-testid="search-submit"
              type="button"
            >
              SEARCH
            </button>
          </div>
        </div>
        <div
          class="result-view-wrapper"
        >
          <p
            class="status"
          />
        </div>
        <div
          class="page-controller"
        >
          <button
            data-testid="prev-page"
            disabled=""
            type="button"
          >
            &lt;
          </button>
          <span
            data-testid="page-1"
          >
            Page 
            1
          </span>
          <button
            data-testid="next-page"
            disabled=""
            type="button"
          >
            &gt;
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
